src/libstd/sys/sgx/net.rs

```diff
@@ -103,24 +103,22 @@ impl TcpStream {
         self.inner.inner.read(buf)
     }
 
-    pub fn read_vectored(&self, buf: &mut [IoVecMut<'_>]) -> io::Result<usize> {
-        let buf = match buf.get_mut(0) {
-            Some(buf) => buf,
-            None => return Ok(0),
-        };
-        self.read(buf)
+    pub fn read_vectored(&self, bufs: &mut [IoVecMut<'_>]) -> io::Result<usize> {
+        match bufs.iter_mut().find(|b| !b.is_empty()) {
+            Some(buf) => self.read(buf),
+            None => Ok(0),
```

Comment:
Is this correct? On at least Linux & Windows, `read(&mut [])` blocks until the fd is ready for reading.

Reply:
Sure, we can change the default behavior to fall back to an empty buffer read.